lib/lte_link_control/modules/enveval.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <zephyr/logging/log.h>
+#include <modem/at_monitor.h>
+#include <modem/lte_lc.h>
+#include <modem/lte_lc_trace.h>
+#include <modem/at_parser.h>
+#include <nrf_modem_at.h>
+
+#include "common/event_handler_list.h"
+#include "common/helpers.h"
+#include "modules/cfun.h"
+#include "modules/enveval.h"
+
+LOG_MODULE_DECLARE(lte_lc, CONFIG_LTE_LINK_CONTROL_LOG_LEVEL);
+
+/* ENVEVAL AT commands */
+#define AT_ENVEVAL		"AT%%ENVEVAL=%d,%s"
+#define AT_ENVEVAL_CANCEL	"AT%%ENVEVALSTOP"
+
+/** "MCCMNC" (including quotes) + comma or null termination */
+#define PLMN_ENTRY_SIZE	9
+
+/* Global variable to accumulate environment evaluation results */
+static struct lte_lc_env_eval_result env_eval_result_accumulator;
+static bool env_eval_in_progress;
```

Comment:
We need to ensure there is no corruption so that timings cannot cause a new event to override the data while we are dispatching the event. 

After writing the above, I notice that these variables are only accessed from `at_handler_enveval`. Should these be moved in there?

Reply:
Yes, I will move those.